helm/cluster-gcp/templates/_control_plane.tpl

```diff
@@ -20,14 +20,128 @@ serviceAccounts:
+    extraVolumes:
+    - name: auditlog
+      hostPath: /var/log/apiserver
+      mountPath: /var/log/apiserver
+      readOnly: false
+      pathType: DirectoryOrCreate
+    - name: policies
+      hostPath: /etc/kubernetes/policies
+      mountPath: /etc/kubernetes/policies
+      readOnly: false
+      pathType: DirectoryOrCreate
+    - name: encryption
+      hostPath: /etc/kubernetes/encryption
+      mountPath: /etc/kubernetes/encryption
+      readOnly: false
+      pathType: DirectoryOrCreate
+  controllerManager:
+    extraArgs:
+      allocate-node-cidrs: "true"
+      authorization-always-allow-paths: "/healthz,/readyz,/livez,/metrics"
+      bind-address: 0.0.0.0
+      cloud-provider: gce
+      cloud-config: /etc/kubernetes/gcp.conf
+      cluster-cidr: {{ join "," .Values.network.podCidr }}
+      logtostderr: "true"
+      profiling: "false"
+    extraVolumes:
+    - hostPath: /etc/kubernetes/gcp.conf
+      mountPath: /etc/kubernetes/gcp.conf
+      name: cloud-config
+      readOnly: true
+  scheduler:
+    extraArgs:
+      authorization-always-allow-paths: "/healthz,/readyz,/livez,/metrics"
+      bind-address: 0.0.0.0
+  etcd:
+    local:
+      imageRepository: {{.Values.controlPlane.etcd.imageRepository}}
+      imageTag: {{.Values.controlPlane.etcd.imageTag}}
+      extraArgs:
+        listen-metrics-urls: "http://0.0.0.0:2381"
+        quota-backend-bytes: "8589934592"
+  networking:
+    serviceSubnet: {{ .Values.network.serviceCIDR }}
+files:
+{{- include "sshFiles" . | nindent 4 }}
+{{- include "diskFiles" . | nindent 4 }}
+{{- include "kubeProxyFiles" . | nindent 4 }}
+{{- include "kubernetesFiles" . | nindent 4 }}
+initConfiguration:
+  skipPhases:
+  - addon/kube-proxy
+  localAPIEndpoint:
+    advertiseAddress: ""
+    bindPort: 0
+  nodeRegistration:
+    kubeletExtraArgs:
+      cloud-provider: gce
+      healthz-bind-address: 0.0.0.0
+      image-pull-progress-deadline: 1m
+      node-ip: '{{ `{{ ds.meta_data.local_ipv4 }}` }}'
+      v: "2"
+    name: '{{ `{{ ds.meta_data.local_hostname.split(".")[0] }}` }}'
+joinConfiguration:
+  discovery: {}
+  nodeRegistration:
+    kubeletExtraArgs:
+      cloud-provider: gce
+    name: '{{ `{{ ds.meta_data.local_hostname.split(".")[0] }}` }}'
+preKubeadmCommands:
+- /bin/bash /opt/init-disks.sh
+- /bin/bash /etc/gs-kube-proxy-patch.sh
+postKubeadmCommands:
+{{- include "sshPostKubeadmCommands" . | nindent 4 }}
+users:
+{{- include "sshUsers" . | nindent 4 }}
+{{- end }}
+
 
 {{- define "control-plane" }}
 apiVersion: controlplane.cluster.x-k8s.io/v1beta1
 kind: KubeadmControlPlane
 metadata:
   labels:
     {{- include "labels.common" $ | nindent 4 }}
-  name: {{ include "resource.default.name" $ }}
+  name: {{ include "resource.default.name" $ }}-{{ include "hash" (dict "data" (include "controlplane-kubeadmcontrolplane-spec" $) "global" .) }}
```

Comment:
You need to update the name here too: https://github.com/giantswarm/cluster-gcp/blob/main/helm/cluster-gcp/templates/_cluster.tpl#L19

Reply:
done, thanks!